azkaban-common/src/main/java/azkaban/executor/container/watch/FlowStatusManagerListener.java

```diff
@@ -50,7 +55,7 @@
  * (2) Driving pod lifecycle actions, such as deleting flow pods in a final state.
  */
 @Singleton
-public class FlowStatusManagerListener implements AzPodStatusListener {
+public class FlowStatusManagerListener extends EventHandler implements AzPodStatusListener {
```

Comment:
Why does this need to extend the `EventHandler`?
Please see if equivalent functionality can be provided by:
- Passing an EventHandler or EventListener object to this class and retaining that as a member variable
- Using that handler directly whenever we need to emit an event, instead of extending `FlowStatusManagerListener` directly.
 As a general principle composition should be favored over inheritance: https://en.wikipedia.org/wiki/Composition_over_inheritance (unless the class hierarchy arises as a more natural representation of the use case)